File: api/enums/scan_types.py
import graphene


class ScanTypeEnums(graphene.Enum):
    MAIL = "mail"
    WEB = "web"

    @property
    def description(self):
        if self == ScanTypeEnums.MAIL:
            return "Used for defining if DMARC and DKIM scans should be performed"
        elif self == ScanTypeEnums.WEB:
            return "Used for defining if HTTPS and SSL scans should be performed"
        else:
            return "Error, this enum value does not exist."

File: api/src/enums/scan-types.js
import { GraphQLEnumType } from 'graphql'

export const ScanTypeEnums = new GraphQLEnumType({
  name: 'ScanTypeEnums',
  values: {
    MAIL: {
      value: 'mail',
      description:
        'Used for defining if DMARC and DKIM scans should be preformed.',
    },
    WEB: {
      value: 'web',
      description:
        'Used for defining if HTTPS and SSL scans should be preformed.',
    },
  },
  description:
    'Enum used when requesting a manual scan to determine what type of scan is to be ran.',
})
